inject/src/main/java/io/micronaut/inject/InstantiatableBeanDefinition.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.inject;
+
+import io.micronaut.context.BeanContext;
+import io.micronaut.context.BeanResolutionContext;
+import io.micronaut.context.DefaultBeanResolutionContext;
+import io.micronaut.context.exceptions.BeanInstantiationException;
+
+/**
+ * <p>An type of {@link BeanDefinition} that can build a new instance.</p>
+ *
+ * @param <T> The bean type
+ * @author Denis Stepanov
+ * @since 4.0
+ */
+public interface InstantiatableBeanDefinition<T> extends BeanDefinition<T> {
+
+    /**
+     * Builds a bean instance.
+     *
+     * @param context    The context
+     * @return The instance
+     * @throws BeanInstantiationException if the instance could not be instantiated
+     */
+    default T instantiate(BeanContext context) throws BeanInstantiationException {
+        return instantiate(new DefaultBeanResolutionContext(context, this), context);
```

Comment:
should be wrapped in try with resources to that inject scoped beans are destroyed